packages/dds/tree/src/simple-tree/core/getOrCreateNode.ts

```diff
@@ -5,7 +5,6 @@
 
 import type { TreeValue } from "../../core/index.js";
 import { isFlexTreeNode, type FlexTreeUnknownUnboxed } from "../../feature-libraries/index.js";
-import type { TreeLeafValue } from "../schemaTypes.js";
```

Comment:
ideally core would not import things from top level in simple-tree, and those files use core. With this change, these undesired deps are removed.